src/celmodel/mesh.h

```diff
@@ -8,144 +8,151 @@
-                        VertexAttributeFormat _format,
-                        unsigned int _offset) :
-            semantic(_semantic),
-            format(_format),
-            offset(_offset)
-        {
-        }
-
-        VertexAttributeSemantic semantic;
-        VertexAttributeFormat   format;
-        unsigned int            offset;
-    };
+    }
+
+    VertexAttribute(VertexAttributeSemantic _semantic,
+                    VertexAttributeFormat _format,
+                    unsigned int _offset) :
+        semantic(_semantic),
+        format(_format),
+        offset(_offset)
+    {
+    }
+
+    VertexAttributeSemantic semantic;
+    VertexAttributeFormat   format;
+    unsigned int            offset;
+};
+
 
-    struct VertexDescription
+struct VertexDescription
+{
+    VertexDescription(unsigned int _stride,
+                        unsigned int _nAttributes,
+                        VertexAttribute* _attributes);
+    VertexDescription(const VertexDescription& desc);
+    ~VertexDescription();
+
+    inline const VertexAttribute& getAttribute(VertexAttributeSemantic semantic) const
     {
-        VertexDescription(unsigned int _stride,
-                          unsigned int _nAttributes,
-                          VertexAttribute* _attributes);
-        VertexDescription(const VertexDescription& desc);
-        ~VertexDescription();
+        return semanticMap[static_cast<std::size_t>(semantic)];
+    }
 
-        const VertexAttribute& getAttribute(VertexAttributeSemantic semantic) const
-        {
-            return semanticMap[semantic];
-        }
+    VertexDescription appendingAttributes(const VertexAttribute* newAttributes, int count) const;
 
-        VertexDescription appendingAttributes(const VertexAttribute* newAttributes, int count) const;
+    bool validate() const;
 
-        bool validate() const;
+    VertexDescription& operator=(const VertexDescription&);
 
-        VertexDescription& operator=(const VertexDescription&);
+    unsigned int stride;
+    unsigned int nAttributes;
+    VertexAttribute* attributes;
 
-        unsigned int stride;
-        unsigned int nAttributes;
-        VertexAttribute* attributes;
+private:
+    void clearSemanticMap();
+    void buildSemanticMap();
 
-    private:
-        void clearSemanticMap();
-        void buildSemanticMap();
+    // Vertex attributes indexed by semantic
+    std::array<VertexAttribute, static_cast<std::size_t>(VertexAttributeSemantic::SemanticMax)> semanticMap;
+};
 
-        // Vertex attributes indexed by semantic
-        VertexAttribute semanticMap[SemanticMax];
-    };
 
-    enum PrimitiveGroupType
-    {
-        TriList    = 0,
-        TriStrip   = 1,
-        TriFan     = 2,
-        LineList   = 3,
-        LineStrip  = 4,
-        PointList  = 5,
-        SpriteList = 6,
-        PrimitiveTypeMax = 7,
-        InvalidPrimitiveGroupType = -1
-    };
+class PrimitiveGroup
+{
+public:
+    PrimitiveGroup() = default;
```

Comment:
these 2 are uneeded

Reply:
They were present in the class when it was nested and I knew I would need them again in the next commit, so I didn't bother removing them here. If you want I can remove them in this commit and re-add them in the next one but I'm not sure there's much point.